assets/js/plan_template.js

```diff
@@ -0,0 +1,175 @@
+// JS for plan_template.html
+import { logOut, updateUsername } from "./user.js";
+document.getElementById("logout-confirm-btn").addEventListener(
+    "click", logOut
+)
+
+const username = updateUsername();
+document.getElementById("profile").addEventListener("click", () => window.location = `/v1/users/${username}/profile`);
+
+function removeLastRow() {
+    const table = document.getElementById("template");
+    // prevent deleting table headers
+    if (table.rows.length > 1) {
+        table.deleteRow(-1);
+    }
+}
+
+function insertNewRow(start = '8', end = '10', category = 'Visit') {
+    const template = document.getElementById("template");
+    let newRow = template.insertRow(-1);
+    let newCell = newRow.insertCell(0);
+    let newCell2 = newRow.insertCell(1);
+    let newCell3 = newRow.insertCell(2);
+
+    let select = document.createElement("select");
+    let cafeOption = document.createElement("option");
+    let attractionOption = document.createElement("option");
+
+    let startTime = hourDropdown();
+    startTime.classList.add("form-select");
+    startTime.value = start;
+
+    let endTime = hourDropdown();
+    endTime.classList.add("form-select");
```

Comment:
Could we also include modifying classList of `endTime` inside the function `hourDropdown()`, similar for the value attr?